### PR TITLE
[primTorch] Enable regex error testing for `masked_fill` ref

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #87767
* #87766
* #87765

cc @ezyang @mruberry @ngimel @Lezcano @peterbell10 @fdrocha

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -4527,8 +4527,6 @@ def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLi
             lambda: "Expected `value` to be on same device as `a`",
         )
         value_type = utils.dtype_to_type(value.dtype)
-        if utils.is_cpu_scalar_tensor(value):
-            value = value.item()
```
Comment (nkaretnikov) [inline torch/_refs/__init__.py:4530]: Is this okay? Looks like it doesn't break anything and more tests pass now (due to not having data-dependent control flow). I've also added a repro that was linked from a PR that added this, see `test_masked_fill_device_promotion`.

Comment (lezcano) [reply torch/_refs/__init__.py:4533]: It makes sense to me. @mruberry @ngimel ?

**Review by lezcano — APPROVED** [review]
LGTM, but let's wait see if Natalia / Mike have anything to say.

File: test/test_torch.py [hunk 2]
```diff
@@ -3703,6 +3703,25 @@ def test_masked_fill_bool_tensor(self, device):
         dst = dst.masked_fill(mask, False)
         self.assertEqual(dst, torch.tensor([True, False, True], device=device))
 
+    # masked_fill() ref used to use item(), make sure the reported repro works.
+    # Not an OpInfo test because it breaks TestMathBitsCUDA::test_conj_view.
+    # https://github.com/pytorch/pytorch/issues/81018 (repro)
+    # https://github.com/pytorch/pytorch/pull/82737 (context)
+    @skipIfRocm
+    @onlyCUDA  # we specifically need device='cuda' for this one
+    def test_masked_fill_device_promotion(self, device):
```
Comment (ngimel) [inline test/test_torch.py:3712]: why can't we add this reference input to OpInfo? TestMathBits doesn't run on reference inputs.

Comment (nkaretnikov) [reply test/test_torch.py:3712]: Ah, I didn't know that! I've only tried putting it in sample inputs. Will do!

**Pull request closed without merging**